gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowStatusResource.java

```diff
@@ -106,14 +107,22 @@ private FlowStatus convertFlowStatus(org.apache.gobblin.service.monitoring.FlowS
     JobStatusArray jobStatusArray = new JobStatusArray();
     FlowId flowId = new FlowId().setFlowName(monitoringFlowStatus.getFlowName())
         .setFlowGroup(monitoringFlowStatus.getFlowGroup());
-    long flowStartTime = Long.MAX_VALUE;
-    long flowEndTime = -1L;
-    // flow execution status is complete unless job status indicates it is running or failed
-    ExecutionStatus flowExecutionStatus = ExecutionStatus.COMPLETE;
+
+    long flowEndTime = 0L;
+    ExecutionStatus flowExecutionStatus = ExecutionStatus.$UNKNOWN;
+
     StringBuffer flowMessagesStringBuffer = new StringBuffer();
 
     while (jobStatusIter.hasNext()) {
       org.apache.gobblin.service.monitoring.JobStatus queriedJobStatus = jobStatusIter.next();
+
+      // Check if this is the flow status instead of a single job status
+      if (queriedJobStatus.getJobName().equals(JobStatusRetriever.NA_KEY) && queriedJobStatus.getJobGroup().equals(JobStatusRetriever.NA_KEY)) {
+        flowEndTime = queriedJobStatus.getEndTime();
```

Comment:
keep the TODO
// TODO: end time should be left as -1 if not all jobs have finished for the flow, i.e. flow status is still RUNNING

Reply:
Or maybe fix, if you can in this PR. (optional)

Reply:
This is already fixed by this PR, the flow end time will now be 0 while running instead of previously where it would show some job's end time.